drivers/i2c/i2c_lpc11u6x.c

```diff
@@ -349,8 +349,8 @@ static int lpc11u6x_i2c_init(const struct device *dev)
 static const struct i2c_driver_api i2c_api = {
 	.configure = lpc11u6x_i2c_configure,
 	.transfer = lpc11u6x_i2c_transfer,
-	.slave_register = lpc11u6x_i2c_slave_register,
-	.slave_unregister = lpc11u6x_i2c_slave_unregister,
+	.target_register = lpc11u6x_i2c_slave_register,
+	.target_unregister = lpc11u6x_i2c_slave_unregister,
 };
```

Comment:
Should we rename the driver functions here as well?

Reply:
good point, I think for consistency we should.

Reply:
Yes, but that could be done gradually over time, the goal here was to correct all the public facing APIs as noted in the PR